pydropsonde/helper/xarray_helper.py

```diff
@@ -28,17 +28,14 @@ def add_ancillary_var(ds, variable, anc_name):
     return ds
 
 
-# encode and write files
-def get_chunks(ds, var, object_dim="sonde_id", alt_dim="alt"):
-    """
-    get standard chunks for one object_dim (like sonde_id or circle_id) and one height dimension
-    """
-    chunks = {
-        object_dim: min(256, ds.sonde_id.size),
-        alt_dim: min(400, ds[alt_dim].size),
-    }
+def get_chunks(ds, var, object_dims=("sonde_id", "circle_id"), alt_dim="alt"):
+    chunks = {alt_dim: min(400, ds[alt_dim].size) if alt_dim in ds.dims else None}
+
+    for dim in object_dims:
+        if dim in ds.dims:
```

Comment:
The get_chunks function could be something like:

```

# encode and write files
def get_chunks(ds, var, object_dims=("sonde_id", "circle_id"), alt_dim="alt"):
    """
    get standard chunks for one object_dim (like sonde_id or circle_id) and one height dimension
    """
    chunks = {
        object_dim: min(256, ds[object_dim].size)
        for object_dim in object_dims
    }
    if alt_dim in ds[var].dims:
        chunks.update(
            {
            alt_dim: min(400, ds[alt_dim].size),
            }
        )
    return tuple((chunks[d] for d in ds[var].dims))

```

if you make sure, that at each point in the code where it is called the argument that is passed to object_dims is actually a tuple (e.g. in "write_l2" it could be `object_dims=("sonde_id",),` instead of `object_dims="sonde_id"`,